### DISCUSSION TITLE
Option to set certain note toolbars as "additive", so their buttons merge into any note toolbar that's mapped higher

### DISCUSSION THREAD

**Discussion by FeralFlora** [chrisgurney/obsidian-note-toolbar#77] [Ideas]
As is, toolbars specified by properties will override folder mappings. However, I think it would be useful to set certain simple / minimal toolbars to merge with any toolbar mapped above it.

For example, I have a global toolbar (mapped to *) with certain common commands I use (go up, create derived note, log something).

In addition, I have a project toolbar that currently only has one command: Log to project. Rather than having the project toolbar override the global toolbar, I would prefer if the "Log to project" command was simply added to the global toolbar, so the two toolbars merge, so to speak. In this sense, the Project toolbar would be additive - it adds to toolbars mapped above it.

**Comment by chrisgurney** [comment]
Thanks for your vote! Take a look at #38 #32 and #34 which both reference this same idea.

**Reply by FeralFlora** [reply]
#38 is slightly different, as it's about having a whole toolbar represented as a button, which then replaces the initial toolbar with another if clicked.

Here's my idea in more detail. If you have two toolbars:

- T1
    - Mapped to `*`
    - Contains buttons b1, b2, b3
- T2
    - Not mapped to a folder, only specified by property
    - Contains buttons b4 and b5
    - Is configured to be "additive" or whatever we should call it
    
If I navigate to a note with the `notetoolbar: T2` property, then the buttons of T2 would be seamlessly added to T1, so it in essence becomes a merged toolbar T1+2 with the buttons b1, b2, b3, b4 and b5.

Of course, the question remains which toolbar takes precedence in terms of styling. I think the natural option is T1.

Hehe, and now I see that this idea was already laid out in #32. I should have gone through all the links before responding...
